wrapanapi/containers/providers/rhopenshift.py

```diff
@@ -150,92 +151,137 @@ def __init__(self, hostname, protocol="https", port=8443, k_entry="api/v1", o_en
-        return entities
+        return [item for item in self.o_api.list_image().items
+                if item.docker_image_reference is not None]
 
-    def list_deployment_config(self):
+    def list_deployment_config(self, namespace=None):
         """Returns list of deployment configs"""
-        entities = []
-        entities_j = self.o_api.get('deploymentconfig')[1]['items']
-        for entity_j in entities_j:
-            meta = entity_j['metadata']
-            entity = DeploymentConfig(self, meta['name'], meta['namespace'])
-            entities.append(entity)
-        return entities
+        if namespace:
+            dc = self.o_api.list_namespaced_deployment_config(namespace=namespace).items
+        else:
+            dc = self.o_api.list_deployment_config_for_all_namespaces().items
+        return dc
+
+    def list_service(self, namespace=None):
+        """Returns list of services."""
+        if namespace:
+            svc = self.k_api.list_namespaced_service(namespace=namespace).items
+        else:
+            svc = self.k_api.list_service_for_all_namespaces().items
+        return svc
+
+    def list_replication_controller(self, namespace=None):
+        """Returns list of replication controllers"""
+        if namespace:
+            rc = self.k_api.list_namespaced_replication_controller(namespace=namespace).items
+        else:
+            rc = self.k_api.list_replication_controller_for_all_namespaces().items
+        return rc
+
+    def list_node(self):
+        """Returns list of nodes"""
+        nodes = self.k_api.list_node().items
+        return nodes
+
+    def cluster_info(self):
+        """Returns information about the cluster - number of CPUs and memory in GB"""
+        aggregate_cpu, aggregate_mem = 0, 0
+        for node in self.list_node():
+            aggregate_cpu += int(node.status.capacity['cpu'])
+            aggregate_mem += int(round(int(node.status.capacity['memory'][:-2]) * 0.00000102400))
+
+        return {'cpu': aggregate_cpu, 'memory': aggregate_mem}
+
+    def list_persistent_volume(self):
+        """Returns list of persistent volumes"""
+        pv = self.k_api.list_persistent_volume().items
+        return pv
+
+    def list_pods(self, namespace=None):
+        """Returns list of container groups (pods).
+        If project_name is passed, only the pods under the selected project will be returned"""
+        if namespace:
+            pods = self.k_api.list_namespaced_pod(namespace=namespace).items
+        else:
+            pods = self.k_api.list_pod_for_all_namespaces().items
+        return pods
+
+    def list_container(self, namespace=None):
+        """Returns list of containers (derived from pods)
+        If project_name is passed, only the containers under the selected project will be returned
+        """
+        pods = self.list_pods(namespace=namespace)
+        containers = []
+        containers.extend([pod.spec.containers for pod in pods])
```

Comment:
Instead of extending the empty list, can we just do this?
`containers = [pod.spec.containers for pod in pods]`

Reply:
there was more code before